src/main/java/io/wispforest/owo/registration/reflect/MemoizedEntry.java

```diff
@@ -0,0 +1,41 @@
+package io.wispforest.owo.registration.reflect;
+
+import net.minecraft.registry.entry.RegistryEntry;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Supplier;
+
+public class MemoizedEntry<T> implements Supplier<T> {
+
```

Comment:
is this class needed if there already is a helper method on `AutoRegistryContainer` for making a `MemoizedRegistryEntry`? any people who only want a supplier can still use the registry entry class since you made it extend supplier